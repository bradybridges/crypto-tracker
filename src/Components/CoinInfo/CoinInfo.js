import React from 'react';
import { NavLink } from 'react-router-dom';
import CoinChangeInfo from '../CoinChangeInfo/CoinChangeInfo';
import { connect } from 'react-redux';
import './CoinInfo.scss';

const CoinInfo = (props) => {
  const { coin } = props;
  const timeFrameData = coin[props.timeFrame];
  return (
    <section>
      <h2>{coin.name}</h2>
      <img src={coin.logo_url} alt='coin logo' />
      <p>{coin.rank}</p>
      <h2>{coin.id}</h2>
      <p>{coin.price}</p>
      <p>{coin.high}</p>
      <p>{coin.circulating_supply}</p>
      <p>{coin.max_supply}</p>
      <p>${coin.market_cap}</p>
      <CoinChangeInfo data={timeFrameData} timeFrame={props.timeFrame}/>
    </section>
  );
}

const mapStateToProps = state => ({
  timeFrame: state.timeFrame,
});

export default connect(mapStateToProps)(CoinInfo);

// currency(pin):"BTC"
// id(pin):"BTC"
// price(pin):"9248.95427101"
// price_date(pin):"2019-11-02"
// symbol(pin):"BTC"
// circulating_supply(pin):"18025025"
// max_supply(pin):"21000000"
// name(pin):"Bitcoin"
// logo_url(pin):"https://s3.us-east-2.amazonaws.com/nomics-api/static/images/currencies/btc.svg"
// market_cap(pin):"166712631958.79"
// rank(pin):"1"
// high(pin):"19408.82122873"
// high_timestamp(pin):"2017-12-16"