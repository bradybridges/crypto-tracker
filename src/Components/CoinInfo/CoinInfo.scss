p {
  width: 90%;
  text-align: center;
}

#coin-info-container {
  background-image: url('../../Images/30\ Black\ abstract\ Wallpapers\ HD\ Download.jpg');
  height: 70vh;
  overflow: scroll;
  display: flex;
  flex-direction: column;
  align-items: center;
  color: white;
}

#coin-title-div {
  width: 90%;
  display: flex;
  justify-content: center;
}

#back-button-home {
  position: absolute;
  top: 22vh;
  left: 20px;
}