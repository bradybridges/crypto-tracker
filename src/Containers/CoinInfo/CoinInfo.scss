p {
  width: 90%;
  text-align: center;
}

.coin-p {
  width: 90%;
  margin: 5px;
  padding: 10px;
  background-color: #3e3e3ed4;
  border-radius: 4px;
  display: flex;
  justify-content: space-between;
}

#coin-info-container {
  background-image: url('../../Images/30\ Black\ abstract\ Wallpapers\ HD\ Download.jpg');
  height: 70vh;
  overflow: scroll;
  display: flex;
  flex-direction: column;
  align-items: center;
  color: white;
}

#coin-title-div {
  width: 90%;
  display: flex;
  justify-content: center;
  margin-top: 20px;
}

#back-button-home {
  position: absolute;
  top: 22vh;
  left: 20px;
  background-color: transparent;
}

#coin-rank-info {
  width: 90%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.coin-main {
  font-size: 1.5em;
  margin: 0;
  font-weight: bold;
}

#coin-market-info {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}