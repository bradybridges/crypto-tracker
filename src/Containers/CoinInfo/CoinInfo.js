import React from 'react';
import { NavLink } from 'react-router-dom';
import CoinChangeInfo from '../../Components/CoinChangeInfo/CoinChangeInfo';
import { connect } from 'react-redux';
import './CoinInfo.scss';
import backButton from '../../Images/left-arrow.svg';

export const CoinInfo = (props) => {
  const { coin } = props;
  const timeFrameData = coin[props.timeFrame];
  const price = Number(coin.price).toFixed(2);
  const high = Number(coin.high).toFixed(2);
  return (
    <section id='coin-info-container'>
      <NavLink to='/' id='back-button-home'>
        <img src={backButton} alt='back button' />
      </NavLink>
      <div id='coin-title-div'>
        <h2>{coin.name}</h2>
        <img src={coin.logo_url} alt='coin logo' />
      </div>
      <p>Rank: #{coin.rank}</p>
      <p>Symbol: {coin.id}</p>
      <p>Price: ${price}</p>
      <p>Market Cap: ${coin.market_cap}</p>
      <p>High: ${high}</p>
      <p>Circulating Supply: {coin.circulating_supply}</p>
      <p>Max Supply: {coin.max_supply}</p>
      <CoinChangeInfo data={timeFrameData} timeFrame={props.timeFrame}/>
    </section>
  );
}

export const mapStateToProps = state => ({
  timeFrame: state.timeFrame,
});

export default connect(mapStateToProps)(CoinInfo);