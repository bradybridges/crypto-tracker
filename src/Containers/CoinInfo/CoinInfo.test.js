import React from 'react';
import { shallow } from 'enzyme';
import { CoinInfo, mapStateToProps } from './CoinInfo';

describe('CoinInfo', () => {
  const mockCoin = {
    name: 'Bitcoin',
    logo_url: 'https://s3.us-east-2.amazonaws.com/nomics-api/static/images/currencies/btc.svg',
    rank: '1',
    id: 'BTC',
    price: '9393.23',
    high: '19567.43',
    circulating_supply: '12314534',
    max_supply: '200000000',
    market_cap: '234453454.34',
    '1d': {
      price_change: '231.34',
    }
  }
  const wrapper = shallow(<CoinInfo coin={mockCoin} timeFrame='1d' />)

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should return timeFrame', () => {
    const mockState = {
      timeFrame: '1d',
      error: 'An error occured',
    };
    const expected = { timeFrame: '1d' };
    const result = mapStateToProps(mockState);
    expect(result).toEqual(expected);
  });
});