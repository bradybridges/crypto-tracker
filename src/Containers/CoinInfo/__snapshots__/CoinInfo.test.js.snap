// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinInfo should match the snapshot 1`] = `
<section
  id="coin-info-container"
>
  <NavLink
    id="back-button-home"
    to="/"
  >
    <img
      alt="back button"
      src="left-arrow.svg"
    />
  </NavLink>
  <div
    id="coin-title-div"
  >
    <h2>
      Bitcoin
    </h2>
    <img
      alt="coin logo"
      src="https://s3.us-east-2.amazonaws.com/nomics-api/static/images/currencies/btc.svg"
    />
  </div>
  <p>
    Rank: #
    1
  </p>
  <p>
    Symbol: 
    BTC
  </p>
  <p>
    Price: $
    9393.23
  </p>
  <p>
    Market Cap: $
    234453454.34
  </p>
  <p>
    High: $
    19567.43
  </p>
  <p>
    Circulating Supply: 
    12314534
  </p>
  <p>
    Max Supply: 
    200000000
  </p>
  <CoinChangeInfo
    data={
      Object {
        "price_change": "231.34",
      }
    }
    timeFrame="1d"
  />
</section>
`;
